### PR TITLE
Add String.indexOf() minibench

### PR DESCRIPTION
**Pull request by dhong44**
Add minibench for String.indexOf()

Signed-off-by: Daniel Hong <daniel.hong@live.com>

### CODE REVIEW

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.ibm.bumblebench.core.MicroBench;
+
+import java.util.Random;
+import java.lang.StringBuilder;
+
+/**
+ * 
+ * on z13 and newer z/Architectures with vector facilities enabled, string
+ * indexOf can be accelerated by vector instructions.
```
Comment (fjeremic) [inline net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: This is not only a Z statement. It is true for x86 as well as Power. I think we should take the "platform" out of this statement and just say that in general indexOf operation can benefit from vectorization and that this benchmark is meant for testing the performance of such implementations.

Comment (dhong44) [reply net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: 7b127ed

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package com.ibm.bumblebench.string;
```
Comment (smlambert) [inline net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: net.adoptopenjdk.bumblebench - is the package naming structure now that this is contributed to open project.

Please include the [Apache v2 license header](https://github.com/AdoptOpenJDK/bumblebench/blob/master/net/adoptopenjdk/bumblebench/examples/EmptyBench.java#L1-L13) to any new files.

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * */
+
+public final class StringIndexOf extends MicroBench {
+
+	private static final int MAX_ITERATIONS_PER_LOOP = option("maxIterations", 10000000);
+	private static final int INDEX_OF_VALUE = option("indexOfValue", 10);
+	private static int STRING_LENGTH = option("stringLength", 0);
+
+	private static String string;
+	private static char[] possibleChars = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
```
Comment (karianna) [inline net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: I've seen this block repeated across tests...

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+		Random rand = new Random();
+		int length;
+		StringBuilder sb = new StringBuilder(1000);
+
+		rand.setSeed(12345L);
+
+        for (int j = 0; j < STRING_LENGTH; ++j){
+            sb.append(possibleChars[rand.nextInt(possibleChars.length)]);
+		}
```
Comment (karianna) [inline net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: whitespace weirdness..

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	private static int value = 0;
+
+	static {
+		STRING_LENGTH = (STRING_LENGTH == 0) ? INDEX_OF_VALUE + 1 : STRING_LENGTH;
+
+		Random rand = new Random();
+		int length;
+		StringBuilder sb = new StringBuilder(1000);
+
+		rand.setSeed(12345L);
```
Comment (karianna) [inline net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: We don't care how random this is?

**Review by karianna — CHANGES_REQUESTED** [review]

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java (hunk 5 above)
Comment (tellison) [reply net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: I expect, to have a repeatable test, the author wanted a predictable sequence of random numbers.

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java (hunk 3 above)
Comment (dhong44) [reply net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: This block is used in both my open PR, #4, for `StringConversionBench.java` and `StringIndexOf.java`, here, how would you suggest avoiding this duplicated code?

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java (hunk 5 above)
Comment (dhong44) [reply net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: In 851b522 I added an option to make the string pseudorandom. By setting the seed I intended to create a benchmark that could be repeated, but if a user wants the test to be more random they can set the option with `-DBumbleBench.random`

**Review by karianna — APPROVED** [review]

File: net/adoptopenjdk/bumblebench/string/StringIndexOf.java (hunk 3 above)
Comment (karianna) [reply net/adoptopenjdk/bumblebench/string/StringIndexOf.java]: Something like a `StringTestData.java`?

**Review by fjeremic — APPROVED** [review]

**Comment by dhong44** [comment]
Prereq #5

**Pull request merged**